src/library_wasi.js

```diff
@@ -58,6 +58,12 @@ var WasiLibrary = {
         'LANG': lang,
         '_': getExecutableName()
       };
+#if ENVIRONMENT_MAY_BE_NODE && !DETERMINISTIC
+      if (ENVIRONMENT_IS_NODE) {
+        // Under node we mirror then entire outer environment
+        env = process.env;
```

Comment:
this is breaking change, I think.
For example, there could be OS shell variable which would point to files which do not exist on VFS.

Reply:
Also, given that folks can still override with using the `ENV` global in JS, do you think it would be a problem in practice?

Reply:
I am also a little worried about breaking changes here.

Comparing this to the filesystem, by default we are sandboxed, and only use minimal things from the environment as makes sense (or no things, in DETERMINISTIC mode). The user can opt into `NODEFS` or `NODERAWFS` to get direct/unsandboxed access to the outside. Maybe something similar makes sense for the environment?

If that direction sounds relevant, I'm not sure what's best but options could include a new option, or adding a new "node direct" option (for both this and files), or perhaps bundling with `NODE[RAW]FS` somehow.

Reply:
How about in the deault case.. should certain env vars be allowed to pass though by default?  e.g. `LANG` and `LC_*` were specifically requested in #18816.   Is there any reason to not pass those ones through?

Reply:
I think specific ones might be ok on a case by case basis.